### PR TITLE
Added and option to pass customer confidence and IoU params 

### PR DESCRIPTION
**Pull request by ipovalyaev**
While serving custom model via torchhub it would be very useful to have a option to pass also custom confidence and IoU params, which are currently hardcoded. 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhancements to YOLOv5 model creation and custom model loading with configurable NMS thresholds.

### 📊 Key Changes
- Introduced `conf` and `iou` parameters in the `_create` and `custom` functions.
- Adjusted `AutoShape` class to accept custom NMS confidence (`conf`) and IoU (`iou`) thresholds upon initialization.

### 🎯 Purpose & Impact
- ⚙️ **Purpose**: Allows users to specify their own values for NMS confidence and IoU thresholds when creating or loading models, enabling more flexibility and fine-tuning based on specific use cases.
- 🎨 **Impact**: Users can now customize the sensitivity of their model's detection capabilities, potentially improving accuracy and reducing false positives depending on the context of the application.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@ipovalyaev see YOLOv5 **PyTorch Hub** tutorial for setting confidence and IoU threshold:

### Inference Settings
YOLOv5 models contain various inference attributes such as **confidence threshold**, **IoU threshold**, etc. which can be set by:
```python
model.conf = 0.25  # NMS confidence threshold
      iou = 0.45  # NMS IoU threshold
      agnostic = False  # NMS class-agnostic
      multi_label = False  # NMS multiple labels per box
      classes = None  # (optional list) filter by class, i.e. = [0, 15, 16] for COCO persons, cats and dogs
      max_det = 1000  # maximum number of detections per image
      amp = False  # Automatic Mixed Precision (AMP) inference

results = model(imgs, size=320)  # custom inference size
```

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://docs.ultralytics.com/yolov5/tutorials/model_export8)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [TFLite, ONNX, CoreML, TensorRT Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

Good luck 🍀 and let us know if you have any other questions!

**Comment by ipovalyaev** [comment]
Thanks a lot, this works, missed it somehow. Sorry for noise :(
